docs/contracts/storage-layout-specifier.rst

```diff
@@ -0,0 +1,61 @@
+.. index:: ! storage layout specifier, ! layout at
+
+.. _storage-layout-specifier:
+
+************************
+Storage Layout Specifier
```

Comment:
The specifier is the syntax element used to provide it, but we should have a more general name for the concept as a whole. This is how sections are named here.

Maybe this?

```suggestion
Custom Storage Layout
```

`Storage Layout Specification` might be good enough as well. But neither sounds that good overall so feel free to suggest something else.

Reply:
I'm personally fine with both, but if we do decide to change it, we should also change the file name to reflect it.